src/main.rs

```diff
@@ -875,18 +857,129 @@ async fn get_to_dir(get: GetInteractive, out_dir: PathBuf) -> Result<()> {
             )
             .progress_chars("#>-"),
     );
+    pb
+}
 
-    let init_download_progress = |count: u64, missing_bytes: u64| {
-        progress!("{} Downloading ...", style("[3/3]").bold().dim());
-        progress!(
-            "  {} file(s) with total transfer size {}",
-            count,
-            HumanBytes(missing_bytes)
-        );
-        pb.set_length(missing_bytes);
-        pb.reset();
-        pb.set_draw_target(ProgressDrawTarget::stderr());
+fn init_download_progress(pb: &ProgressBar, count: u64, missing_bytes: u64) {
+    progress!("{} Downloading ...", style("[3/3]").bold().dim());
+    progress!(
+        "  {} file(s) with total transfer size {}",
+        count,
+        HumanBytes(missing_bytes)
+    );
+    pb.set_length(missing_bytes);
+    pb.reset();
+    pb.set_draw_target(ProgressDrawTarget::stderr());
+}
+
+/// Get a single file
+async fn get_to_file_single(
+    get: GetInteractive,
+    out_dir: PathBuf,
+    temp_dir: PathBuf,
+) -> Result<()> {
+    let hash = get.hash();
+    progress!("Fetching: {}", Blake3Cid::new(hash));
+    progress!("{} Connecting ...", style("[1/3]").bold().dim());
+
+    let name = Blake3Cid::new(hash).to_string();
+    // range I am missing for the 1 file I am downloading
+    let range = get_missing_range(&get.hash(), name.as_str(), &temp_dir, &out_dir)?;
+    if range.is_all() {
+        tokio::fs::create_dir_all(&temp_dir)
+            .await
+            .context("unable to create directory {temp_dir}")?;
+        tokio::fs::create_dir_all(&out_dir)
+            .await
+            .context("Unable to create directory {out_dir}")?;
+    }
+    let query = RangeSpecSeq::new([range]);
+    let pb = make_download_pb();
+
+    // collection info, in case we won't get a callback with is_root
+    let collection_info = Some((1, 0));
+
+    let request = GetRequest::new(get.hash(), query).into();
+    let response = match get {
+        GetInteractive::Ticket {
+            ticket,
+            keylog,
+            derp_map,
+        } => get::run_ticket(&ticket, request, keylog, derp_map).await?,
+        GetInteractive::Hash { opts, .. } => get::run(request, opts).await?,
+    };
+    let connected = response.next().await?;
+    progress!("{} Requesting ...", style("[2/3]").bold().dim());
+    if let Some((count, missing_bytes)) = collection_info {
+        init_download_progress(&pb, count, missing_bytes);
+    }
+    let ConnectedNext::StartRoot(curr) = connected.next().await? else {
+        anyhow::bail!("Unexpected StartChild or Closing");
+    };
```

Comment:
a let-else in the wild!  nice (/me still learning when they're useful)

Reply:
I use them as replacement for the ugly

```rust
let x = match foo {
  Case1(x) => x,
  _ => return
}
```

stuff